src/tests/java/picard/sam/markduplicates/QuerySortedMarkDuplicatesTest.java

```diff
@@ -42,4 +54,41 @@ protected boolean markUnmappedRecordsLikeTheirMates() {
 
     @Override
     protected AbstractMarkDuplicatesCommandLineProgramTester getTester() { return new QuerySortedMarkDuplicatesTester(); }
+
+    private class ByQualityQuerySortedTester extends AbstractMarkDuplicatesCommandLineProgramTester {
+
+        public ByQualityQuerySortedTester() {
+            super(DuplicateScoringStrategy.ScoringStrategy.SUM_OF_BASE_QUALITIES, SAMFileHeader.SortOrder.queryname);
+        }
+
+        @Override
+        protected CommandLineProgram getProgram() {
+            return new MarkDuplicates();
+        }
+    }
+
+    @DataProvider
+    public Object[][] SameUnclipped5primeEnd() {
```

Comment:
methods should start with a lower case, how about `testStrangeOrderDependentBugVersionDataProvider`?